### ISSUE TITLE
[Feature request] Dark settings when dark keyboard

### ISSUE THREAD

**Issue by KaKi87** [florisboard/florisboard#312]
Hello,

It's nice to have a dark keyboard theme, and it would even be nicer to also have a dark settings theme :)

Thanks !

**Comment by patrickgold** [comment]
There is already a dark Settings theme. Go to Settings > Three-dots > Advanced. There you can choose between light and dark theme.

**Comment by KaKi87** [comment]
Nice, thanks. However, that's not user-friendly...

**Comment by patrickgold** [comment]
It defaults to the system theme (if not otherwise set), so at least not that user-unfriendly. A general preferences overhaul is planned for v0.6.0, which maybe makes the three-dot menu less hidden by integrating its options into the main view.

**Comment by KaKi87** [comment]
Why would the user want to have light settings and dark keyboard or dark settings and light keyboard ?
Shouldn't settings theme and keyboard theme be a single setting ?
Plus, this tablet I'm using is a non-rootable Samsung tablet stuck on Android 6...
Thanks

**Comment by patrickgold** [comment]
Hmm good point still I think letting this option decoupled is a better, because else every time you change the keyboard theme the Settings app would have to restart to properly apply the theme and that's bad UX in my opinion.

**Comment by KaKi87** [comment]
Why would the user want to change the keyboard theme without changing the settings theme ?

**Comment by Glitchy-Tozier** [comment]
> Why would the user want to change the keyboard theme without changing the settings theme ?

- Because the user may enjoy the bright settings but still prefer using a dark keyboard.
- Because the user may want to keep his device bright while having the contrast of a dark keyboard.

I can see your point and I think it may be very comfortable in some situations. However, there are a few things that make the decoupled approach better in my opinion:
- Decoupled gives users more control over what the settings look like.
- What if the user uses custom themes? Should the app turn dark just because the custom green theme is listed as "dark"?
- Even if this is a good thing to implement – which I don't think it is – there are more important issues to address.
- Either way, people can complain. The decoupled approach is the default (as far as I can tell). It's present in OpenBoard, Simple Keyboard, ASK, etc.. So I think decoupled is the more widely known and accepted solution and the implementation we should stick to.

**Comment by KaKi87** [comment]
> Decoupled gives users more control over what the settings look like.

Would it be possible, then, to have settings theme configuration right below keyboard theme configuration, in addition to an unchecked by default "use keyboard theme as settings theme" checkbox ?

> What if the user uses custom themes?

Wouldn't the custom theme be applied to settings too ?

> there are more important issues to address

I hope autocomplete is one of those :p

I agree that my issue is low priority and I have no problem with that.

> I think decoupled is the more widely known and accepted solution

... among the FLOSS community. You're omitting that there is only a few user experience designers among us.

**Comment by Glitchy-Tozier** [comment]
> I hope autocomplete is one of those :p

Yup, that's one of the issues I'm looking foward to.  :)

Unfortunately I manage to distract Patrick very efficiently by continuously breaking his app...

>... among the FLOSS community. You're omitting that there is only a few user experience designers among us.

I just didn't want to mention other non-FLOSS keyboards as I have no experience with BIG keyboards like gboard or Swiftkey and have no idea how they handle this.

I assume they don't show this behaviour either, but I may well be wrong about that.

**Comment by kj7rrv** [comment]
They are decoupled in Gboard.

**Comment by patrickgold** [comment]
> Would it be possible, then, to have settings theme configuration right below keyboard theme configuration, in addition to an unchecked by default "use keyboard theme as settings theme" checkbox ?

Technically it is possible, but all Settings Activity related preferences are placed within the Advanced tab and I personally just like it more this way.

> Wouldn't the custom theme be applied to settings too ?

Nope, the Settings app has a static light and dark theme which is pre-compiled and only these two themes can be used to theme the Settings UI components. The keyboard itself uses a custom theme logic to have runtime custom themes.

> I hope autocomplete is one of those :p

Yup :)

> ... among the FLOSS community. You're omitting that there is only a few user experience designers among us.

Gboard does not even give me an option to change the theme on my personal Android 10 device (but it seems to have this option for other devices), it just forces me to take the system theme within their Settings. Changing the keyboard theme in Gboard does not change it either. So this decoupled option is already an improvement.

I will let this issue open for now and assign it to the Backlog. I will probably address this further with the Settings rewrite in v0.6.0.

**Comment by lm41** [comment]
The new keyboard theme editor makes it very complicated as you can specify arbitrary colors for every element of the keyboard and it would be a pain in the ass to match the settings color scheme this way. As of today you can specify the light/dark/amoled color scheme and with #2750 also the accent color of the settings app. I am in favor of closing this issue. CC @patrickgold

**Comment by patrickgold** [comment]
Yes I'd also say closing this issue is the best path forward, the current way of cobtrolling the settings theme is sufficient.
